### PR TITLE
S.O.G. Prairie Fire Compatibility

### PR DESCRIPTION
**Pull request by BaerMitUmlaut**
**When merged this pull request will:**
- Add a compatibility addon for the S.O.G. Prairie Fire CDLC

This is a joint effort of @veteran29 and me. Some highlights:
- Allows you to dig spiderholes with the ACE entrenching tool
- Melee attacks do proper stab and punch damage
- Punji traps create some nasty stab wounds on your legs (and body, for the higher reaching ones)

### CODE REVIEW

File: optionals/compat_sog/CfgAmmo/explosives.hpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
+class DirectionalBombBase;
+class vn_mine_m18_ammo: DirectionalBombBase {
+    EGVAR(Explosives,defuseObjectPosition)[] = {0, 0, 0.15};
```
Comment (mharis001) [inline optionals/compat_sog/CfgAmmo/explosives.hpp:3]: I think `Explosives` should be all lowercase.

Comment (BaerMitUmlaut) [reply optionals/compat_sog/CfgAmmo/explosives.hpp]: Fixed, also fixed an uppercase `EGVAR(minedetector,detectable)`. Didn't find any other occurances with regex.

File: optionals/compat_sog/CfgVehicles.hpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+class CfgVehicles {
+	#include "CfgVehicles\boxes.hpp"
+    #include "CfgVehicles\explosives.hpp"
+	#include "CfgVehicles\helicopters.hpp"
+	#include "CfgVehicles\planes.hpp"
+	#include "CfgVehicles\tracked.hpp"
+	#include "CfgVehicles\turrets.hpp"
+	#include "CfgVehicles\units.hpp"
+	#include "CfgVehicles\wheeled.hpp"
```
Comment (mharis001) [inline optionals/compat_sog/CfgVehicles.hpp:22]: Inconsistent indentation.

File: optionals/compat_sog/CfgVehicles/units.hpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+class Man;
+class CAManBase: Man {
+    class ACE_SelfActions {
+        class ACE_Equipment {
+            class GVAR(digSpiderhole): EGVAR(trenches,digEnvelopeSmall) {
```
Comment (mharis001) [inline optionals/compat_sog/CfgVehicles/units.hpp:5]: Missing base class.
```
ErrorMessage: File z\ace\addons\compat_sog\CfgVehicles\units.hpp, line 4: /CfgVehicles/CAManBase/ACE_SelfActions/ACE_Equipment.ace_compat_sog_digSpiderhole: Undefined base class 'ace_trenches_digEnvelopeSmall'
```

File: optionals/compat_sog/CfgVehicles.hpp (hunk 2 above)
Comment (BaerMitUmlaut) [reply optionals/compat_sog/CfgVehicles.hpp:22]: Oh, there's some tabs in there. Shouldn't CI have caught this?

Comment (jonpas) [reply optionals/compat_sog/CfgVehicles.hpp:22]: Do we check that in ACE3 CI? I don't think so.

Comment (jokoho48) [reply optionals/compat_sog/CfgVehicles.hpp:22]: we do but only in the addons folder and not the optionals

**Comment by Dahlgren** [comment]
`tools/armake` and `tools/x.py` should be removed befor merge I guess?

**Comment by BaerMitUmlaut** [comment]
Ah damnit, accidentally added that.

**Comment by Dahlgren** [comment]
Probably needs dummy files for anything inside VN pbos for pboproject users as well

**Comment by jonpas** [comment]
@Dahlgren We have dummy files in `tools/pDummies` as they aren't required for HEMTT (which `include` is mostly meant for).

File: optionals/compat_sog/CfgWeapons/launchers.hpp [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+};
+class vn_rpg2: vn_Launcher_Base_F {
+    ace_reloadlaunchers_enabled = 1;
+};
+class vn_rpg7: vn_Launcher_Base_F {
+    ace_reloadlaunchers_enabled = 1;
+};
+class vn_sa7: vn_Launcher_Base_F {
+    ace_reloadlaunchers_enabled = 1;
+};
```
Comment (jonpas) [inline optionals/compat_sog/CfgWeapons/launchers.hpp:13]:
```suggestion
};

```

File: optionals/compat_sog/CfgWeapons/helmets.hpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+};
+class vn_o_helmet_zsh3_01: vn_o_headgear_base {
+    HEARING_PROTECTION_VICCREW;
+};
+class vn_o_helmet_zsh3_02: vn_o_helmet_zsh3_01 {
+    ACE_Protection = 1;
+};
+class vn_o_helmet_shl61_01: vn_o_headgear_base {
+    HEARING_PROTECTION_VICCREW;
+};
```
Comment (jonpas) [inline optionals/compat_sog/CfgWeapons/helmets.hpp:37]:
```suggestion
};

```

**Review by jonpas — CHANGES_REQUESTED** [review]

**Review by jonpas — APPROVED** [review]

File: optionals/compat_sog/CfgVehicles/helicopters.hpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
+// UH-1C - https://en.wikipedia.org/wiki/Bell_UH-1_Iroquois
+class vn_helicopter_base;
+class vn_air_uh1_01_base: vn_helicopter_base {
+    EGVAR(refuel,fuelCapacity) = 916;
+};
+
+// UH-1B - https://www.skytamer.com/Bell_H-1B(UH).html
+class vn_air_uh1c_01_base;
+class vn_air_uh1c_07_base: vn_air_uh1c_01_base {
```
Comment (mharis001) [inline optionals/compat_sog/CfgVehicles/helicopters.hpp:9]: UBC:

```
Updating base class vn_air_uh1c_doorguns_base->vn_air_uh1c_01_base, by z\ace\addons\compat_sog\config.cpp/CfgVehicles/vn_air_uh1c_07_base/ (original vn\air_f_vietnam_c\config.bin)
```

File: optionals/compat_sog/CfgWeapons/launchers.hpp [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class vn_Launcher_Base_F;
+class vn_m127: vn_Launcher_Base_F {
+    ace_overpressure_damage = 0;
```
Comment (mharis001) [inline optionals/compat_sog/CfgWeapons/launchers.hpp:3]: Can use `EGVAR` here and below.

File: optionals/compat_sog/CfgVehicles/helicopters.hpp (hunk 6 above)
Comment (PabstMirror) [reply optionals/compat_sog/CfgVehicles/helicopters.hpp:9]:
```suggestion
class vn_air_uh1c_doorguns_base;
class vn_air_uh1c_07_base: vn_air_uh1c_doorguns_base {
```

**Review by PabstMirror — APPROVED** [review]

**Review by mharis001 — APPROVED** [review]

**Review by veteran29 — COMMENTED** [review]
No UBCs with new SOG:PF update.

**Pull request merged**